lychee-lib/src/extract/html5gum.rs

```diff
@@ -173,6 +187,7 @@ impl Emitter for &mut LinkExtractor {
 
     fn emit_current_tag(&mut self) {
         self.flush_old_attribute();
+        self.current_element_nofollow = false;
```

Comment:
I would reset this in `init_start_tag/init_end_tag`

Reply:
I tried that initially, but it didn't work. It would not properly reset the state and instead found the links which should be excluded. If you have the time you can test yourself. Could be that the error is on my end.

Reply:
will try, is there a testcase for this?

Reply:
Yup, the one from this PR.

Reply:
hmm this patch works for me:

```diff
diff --git a/lychee-lib/src/extract/html5gum.rs b/lychee-lib/src/extract/html5gum.rs
index 55d2947..92e13ec 100644
--- a/lychee-lib/src/extract/html5gum.rs
+++ b/lychee-lib/src/extract/html5gum.rs
@@ -172,13 +172,12 @@ impl Emitter for &mut LinkExtractor {
     fn init_start_tag(&mut self) {
         self.flush_current_characters();
         self.current_element_name.clear();
+        self.current_element_nofollow = false;
         self.current_element_is_closing = false;
     }

     fn init_end_tag(&mut self) {
-        self.flush_current_characters();
-        self.current_element_name.clear();
-        self.current_element_is_closing = true;
+        self.init_start_tag();
     }

     fn init_comment(&mut self) {
@@ -187,7 +186,6 @@ impl Emitter for &mut LinkExtractor {

     fn emit_current_tag(&mut self) {
         self.flush_old_attribute();
-        self.current_element_nofollow = false;
     }

     fn emit_current_doctype(&mut self) {}
```